packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -632,11 +641,14 @@ export class GarbageCollector implements IGarbageCollector {
         },
     ): Promise<IGCStats> {
         const {
-            logger = this.mc.logger,
             runSweep = this.shouldRunSweep,
             fullGC = this.gcOptions.runFullGC === true || this.summaryStateNeedsReset,
         } = options;
 
+        const logger = options.logger
+            ? ChildLogger.create(options.logger, undefined, { all: { gcRunCount: () => this.gcRunCount } })
```

Comment:
What does this exactly do?

Reply:
It will add the properties under `all` (gcRunCount) to all events logged by this logger.